### ISSUE TITLE
Bug when creating a chart on grouped rows/pivot with footer totals

### ISSUE THREAD

**Issue by nuschk** [ag-grid/ag-grid#3822]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

I'm evaluating the enterprise version of ag-grid, but am not yet a customer.

**Providing a Reproducible Scenario**
https://plnkr.co/edit/HPP43gjZGdlCpk3d

**Current behavior**
When trying to create a chart from data which includes total footers (`groupIncludeTotalFooter={ true }`), there is the following error and the chart is not displayed.

<img width="413" alt="grafik" src="https://user-images.githubusercontent.com/5167117/81382977-8e877380-910f-11ea-9b75-965b6ded891d.png">

```
Uncaught TypeError: Cannot read property 'key' of undefined
    at ChartDatasource.getGroupLabels (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9521)
    at eval (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9414)
    at Array.forEach (<anonymous>)
    at _loop_1 (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9405)
    at ChartDatasource.extractRowsFromGridRowModel (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9462)
    at ChartDatasource.getData (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9385)
    at ChartDataModel.updateData (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9806)
    at ChartDataModel.updateCellRanges (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9601)
    at ChartDataModel.init (unpkg.com/ag-grid-enterprise@23.1.0/dist/ag-grid-enterprise.cjs.js:9585)
    at eval (unpkg.com/ag-grid-community@23.1.0/dist/ag-grid-community.cjs.js:3584)
```

The same thing happens for grouped rows, when the range includes the totals.

```jsx
function GridExample() {
  const rowData = [
    { number: 10000, status: 'act' },
    { number: 20000, status: 'act' },
    { number: 30000, status: 'act' },
    { number: 40000, status: 'act' },
    { number: 50000, status: 'act' },
    { number: 10000, status: 'rem' },
    { number: 20000, status: 'rem' },
    { number: 30000, status: 'rem' },
    { number: 40000, status: 'rem' },
    { number: 10000, status: 'rem' },
    { number: 20000, status: 'rem' },
    { number: 30000, status: 'rem' },
  ];

  const columnDefs = [
    {
      field: 'status',
      enableRowGroup: true,
      enablePivot: true,
      rowGroup: true,
    },
    {
      field: 'number',
      enableRowGroup: true,
      enablePivot: true,
      aggFunc: 'sum',
    },
  ];

  const handleFirstDataRendered = (event) => {
    event.api.createPivotChart({ chartType: 'groupedColumn' });
  };

  return (
    <div className="ag-theme-alpine" style={ { width: '800px', height: '800px' } }>
      <AgGridReact
        rowData={ rowData }
        columnDefs={ columnDefs }
        enableCharts
        enablePivot
        pivotMode
        sideBar
        onFirstDataRendered={ handleFirstDataRendered }

        // Chart will be rendered if this is commented out.
        groupIncludeTotalFooter
      />
    </div>
  );
}
```

**Expected behavior**
The chart should be created, showing the totals as well.

**Please tell us about your environment:**
MacOS Mojave.

* **ag-Grid version:** X.X.X
Enterprise 23.1.0

* **Browser:**
Firefox: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
Chrome: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES6

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by LouisMoore-agGrid** [comment]
Hi,

Thank you for your email. We are aware of this issue and are tracking it in our backlog:

You can follow the status of feature requests, bugs and releases using our pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

If you are an ag-Grid Enterprise Customer, please raise this in our Zendesk support system - contact info@ag-grid.com for access. This is our primary channel for Support.


This is the reference and summary of the relevant issue: AG-4153 - Charting (including Pivot charts) a group column + without a series fails if using groupIncludeTotalFooter

**Comment by nuschk** [comment]
👍 , thank you.

**Comment by adamRakAbb** [comment]
Is there any workaround for now?

**Comment by nuschk** [comment]
> Is there any workaround for now?

Hmm, it's been a long time, I don't remember it well. IIRC, we went around and did the pivot ourselves in plain JS. We didn't want to display the grid anyway (just the chart), so that fit our use case well.

The referenced zendesk issue (4153) was closed though (in 24.0.0), so it should be fine I guess.
